sdk/storage/azure-storage-file-datalake/azure/storage/file/datalake/_data_lake_directory_client.py

```diff
@@ -0,0 +1,512 @@
+            For example, if p is 0777 and u is 0057, then the resulting permission is 0720.
+            The default permission is 0777 for a directory and 0666 for a file. The default umask is 0027.
+            The umask must be specified in 4-digit octal notation (e.g. 0766).
+        :keyword str permissions: Optional and only valid if Hierarchical Namespace
+         is enabled for the account. Sets POSIX access permissions for the file
+         owner, the file owning group, and others. Each class may be granted
+         read, write, or execute permission.  The sticky bit is also supported.
+         Both symbolic (rwxrw-rw-) and 4-digit octal notation (e.g. 0766) are
+         supported.
+        :keyword ~datetime.datetime if_modified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only
+            if the resource has been modified since the specified time.
+        :keyword ~datetime.datetime if_unmodified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only if
+            the resource has not been modified since the specified date/time.
+        :keyword str if_match:
+            An ETag value, or the wildcard character (*). Specify this header to perform
+            the operation only if the resource's ETag matches the value specified.
+        :keyword str if_none_match:
+            An ETag value, or the wildcard character (*). Specify this header
+            to perform the operation only if the resource's ETag does not match
+            the value specified. Specify the wildcard character (*) to perform
+            the operation only if the resource does not exist, and fail the
+            operation if it does exist.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: DataLakeDirectoryClient for the subdirectory.
+        """
+        subdir = self.get_sub_directory_client(sub_directory)
+        subdir.create_directory(content_settings=content_settings, metadata=metadata, **kwargs)
+        return subdir
+
+    def delete_sub_directory(self, sub_directory,  # type: Union[DirectoryProperties, str]
+                             **kwargs):
+        # type: (...) -> DataLakeDirectoryClient
+        """
+        Marks the specified subdirectory for deletion.
+
+        :param sub_directory:
+            The directory with which to interact. This can either be the name of the directory,
+            or an instance of DirectoryProperties.
```

Comment:
It looks like this only accepts the name of the directory as a parameter

Reply:
Thanks for reminding me! I made get_sub_directory_client accept both DirectoryProperties and str so that we don't need to touch here. ^_^